src/test/java/io/vertx/core/eventbus/ClusteredEventBusTest.java

```diff
@@ -45,13 +45,19 @@
   @Test
   public void testLocalHandlerNotVisibleRemotely() throws Exception {
     startNodes(2);
+    AtomicInteger received = new AtomicInteger();
+    vertices[1].eventBus().consumer(ADDRESS1).handler(msg -> {
+      if (received.incrementAndGet() == 2) {
+        vertices[1].setTimer(1000, id -> testComplete());
+      }
+    });
     vertices[1].eventBus().localConsumer(ADDRESS1).handler(msg -> {
-      fail("Should not receive message");
+      fail("Local consumer should not receive message");
     });
     vertices[0].eventBus().send(ADDRESS1, "foo");
     vertices[0].eventBus().publish(ADDRESS1, "foo");
-    vertices[0].setTimer(1000, id -> testComplete());
     await();
+    assertEquals(2, received.get());
```

Comment:
It is incorrect from the testing unit perspective: when this statement is reached, `testComplete()` has been called and the test is considered over.

When the test passes it is fine, when the test fails (e.g a regression) then a failure is reported by `AsyncTestBase` that will fail with an `IllegalStateException` instead of the correct error to report.

It can be fixed either by calling junit's `assertEquals` method, or by using a latch without a `testComplete()` to be called

Reply:
Ok I will update it